middleware/request_logs.rb

```diff
@@ -9,12 +9,15 @@ def initialize(app, logger)
       def call(env)
         request = ActionDispatch::Request.new(env)
 
+        # Remove last octet of ip if EU GDPR compliance is needed
+        ip = VCAP::CloudController::Config.config.get(:logging, :gdpr_compliant) ? request.ip.split('.')[0...-1].join('.') + '.0' : request.ip
```

Comment:
This looks like it'll only work with ipv4 - maybe worth looking into a gem that also supports ipv6 to do this as many folks are probably anonymizing IP addresses in ruby for gdpr compliance. (I found https://github.com/ankane/ip_anonymizer after a brief search)

Reply:
I would probably also favor this construct over a ternary:
```ruby
ip = if VCAP::CloudController::Config.config.get(:logging, :gdpr_compliant)
         <redact ip>
      else
        <unredacted-ip>
```

Reply:
@selzoc Thank you for the hint. We changed the implementation quite a bit to allow for v6 adresses. We ditched third party libs to not introduce new dependencies as the task is quite simple to adress with the stdlib of ruby. Also using you recommendation for the variable assignment : ).

If we find we need this in other parts of the CC we will think of pulling the code to anonymize the ip out, somwhere else to make it reusable. For now i think we are good keeping it in request_logs.

Reply:
I still think it would be clearer to pull it out into a function.  But yeah I guess no need for a gem!

Reply:
@selzoc as i am still not that familiar with the codebase, where would such a function be placed in the coding ? Maybe lib/utils ?

Reply:
I just meant extracting lines 15-24 into a function within the same class, just to clean up the `call` function a bit.

Reply:
Done : )